lsp-mode.el

```diff
@@ -4868,18 +4869,53 @@ When language is nil render as markup if `markdown-mode' is loaded."
                  (format "%s (%s)" element count))
           (push element elements))))))
 
+(defface lsp-preferred-code-action-face '((t :underline t))
+  "Face used to highlight preferred code actions.
+Used by `lsp-execute-code-action' and
+`lsp-execute-code-action-by-kind' if they have to prompt for an
+action.")
+
+(defface lsp-disabled-code-action-face '((t :inherit shadow))
+  "Face used to highlight disabled code actions.
+See `lsp-preferred-code-action-face' for details.")
+
+(defface lsp-disabled-code-action-reason-face '((t :inherit lsp-details-face))
+  "Face used to highlight the reason a code action is disabled.
+Shown after the code action in `lsp-execute-code-action',
+... (see `lsp-preferred-code-action-face').")
+
+(lsp-defun lsp--code-action-title ((action &as &CodeAction :title :is-preferred? :disabled?))
+  ;; It would be strange for a code action to be both preferred and disabled, so
+  ;; `cond' is probably good enough.
+  (cond (is-preferred?
+         (propertize title 'face 'lsp-preferred-code-action-face))
+        (disabled?
+         (concat
+          (propertize title 'face 'lsp-disabled-code-action-face)
+          " "
+          (propertize (lsp:code-action-disabled-reason disabled?)
+                      'face 'lsp-disabled-code-action-reason-face)))
+        (t title)))
+
 (defun lsp--select-action (actions)
   "Select an action to execute from ACTIONS."
-  (cond
-   ((seq-empty-p actions) (signal 'lsp-no-code-actions nil))
-   ((and (eq (seq-length actions) 1) lsp-auto-execute-action)
-    (lsp-seq-first actions))
-   (t (let ((completion-ignore-case t))
-        (lsp--completing-read "Select code action: "
-                              (seq-into actions 'list)
-                              (-compose (lsp--create-unique-string-fn)
-                                        #'lsp:code-action-title)
-                              nil t)))))
+  (let ((actions (seq-into actions 'list)))
+    (cond
```

Comment:
If you are going to change it, then change it to pcase.

Reply:
Fixed.